homeassistant/components/mqtt/sensor.py

```diff
@@ -239,7 +239,16 @@ def _update_state(msg: ReceiveMessage) -> None:
                 self._attr_native_value = new_value
                 return
             try:
-                if (payload_datetime := dt_util.parse_datetime(new_value)) is None:
+                try:
+                    new_value_float = float(payload)
+                except ValueError:
+                    new_value_float = None
+
+                if (payload_datetime := dt_util.parse_datetime(new_value)) is None and (
+                    payload_datetime := datetime.fromtimestamp(new_value_float, tz=UTC)
+                    if new_value_float is not None
+                    else None
+                ) is None:
                     raise ValueError
             except ValueError:
                 _LOGGER.warning(
```

Comment:
This does not make sense. If a `ValueError` occurs, we log and return. Lets not nest not blocks.